### ISSUE TITLE
LineIterator is missing from Java bindings

### ISSUE THREAD

**Issue by Leprechaunius** [Itseez/opencv#6095]
Title says it all. I now have to implement my own Bresenham algorithm.

**Comment by Aravind-Suresh** [comment]
I am interested to work on this. Can someone help me how to get started?

**Comment by alalek** [comment]
There is no any sense to wrap LineIterator "as is" into bindings, because almost all methods are C++ specific only. Moreover the main method returns "uchar*" - how do you want to work with pointer in Java?

You can start from usage example in Java for something like this.

**Comment by Leprechaunius** [comment]
Yes, there is no sense in trying to wrap every natively used method, such as `uchar * operator* ()`. 
But that doesn't imply, in any way, the whole class should be omitted. Nobody wants to work with pointers in Java, but there is a `Point pos () const ` function, that returns current pixel as a `Point`. If you need the actual value of the pixel, you can alvays use the `Mat.get()` method, as is always the case with Java bindings.
The other two functions available are just postfix and prefix increment operators, that can be easily replaced with regular methods.
There are a few possible approaches to wrapping this functionality.
For example, it may be as simple as this: https://gist.github.com/Leprechaunius/358d640d5baa5a481a3b
Or one could always just create a static method, that returns a LinkedList of Points - https://gist.github.com/Leprechaunius/57c891faee7e4e1a3ebd
